vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc

```diff
@@ -176,9 +294,174 @@ void MediaTwitter::SavePublisherInfo(
                           duration,
                           window_id,
                           callback);
+
   if (!media_key.empty()) {
     ledger_->SetMediaPublisherInfo(media_key, publisher_key);
   }
 }
 
+void MediaTwitter::OnSaveMediaVisit(
+    ledger::Result result,
+    ledger::PublisherInfoPtr info) {
+}
+
+void MediaTwitter::FetchDataFromUrl(
+    const std::string& url,
+    braveledger_media::FetchDataFromUrlCallback callback) {
+  ledger_->LoadURL(url,
+                   std::vector<std::string>(),
+                   std::string(),
+                   std::string(),
+                   ledger::URL_METHOD::GET,
+                   callback);
+}
+
+void MediaTwitter::OnMediaActivityError(const ledger::VisitData& visit_data,
+                                        uint64_t window_id) {
+  std::string url = TWITTER_TLD;
+  std::string name = TWITTER_MEDIA_TYPE;
+
+  DCHECK(!url.empty());
+
+  ledger::VisitData new_data;
```

Comment:
We should be using base to join url components

Reply:
after discussion we will do it later if needed